src/tapeimage.cpp

```diff
@@ -645,7 +644,7 @@ void tapeimage::read_header_from_disk() noexcept (false) {
             this->recovery = LFP_PROTOCOL_TRYRECOVERY;
             head.prev = back2.next;
         }
-    } else if (this->recovery && !this->index.empty()) {
```

Comment:
You removed empty check.
Now imagine the test: I have a big file with two logical files and open file on the second logical file.
My base address is lets say 1000. But first header that is on position 1000 would be header.prev = 900.
If header type is not 0 or 1, recovery mode will be set, thus the check will fail. 

This check is valid only for the second header.

However saying this, you won't ever see this situation is real life. I actually believe that type =0 or =1 is the strongest check we have and in my opinion it shouldn't be soft fail with recovery, but hard fail. I do not believe think there is any recoverable file for which type is broken.

Reply:
Alright - I believe the solution to that is to re-introduce empty, but empty should not consider ghost nodes.